akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -117,10 +117,10 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
       protocol = c(7) match {
         case '0' => `HTTP/1.0`
         case '1' => `HTTP/1.1`
-        case _   => onBadProtocol
+        case _   => onBadProtocol(input.utf8String)
```

Comment:
What happens if `utf8String` fails?

Reply:
I couldn't find any documentation on `utf8String` throwing any `Exception`, either directly or indirectly by calling `decodeString`.

Reply:
The relevant input only starts at offset `cursor` so that needs to be taken into account. Maybe we should use `LogByteStringTools.printByteString` instead with a sensible limit (32?) to print the invalid data? That might make it easier to find out if someone is attempting to connect to a server that speaks a binary protocol instead of text-y HTTP.

Btw. this is more important at the other usages. At this place in particular we have already seen `HTTP/1.` so we could be a bit more cleverer here?